x-pack/plugins/cases/public/components/user_actions/comment/alert.tsx

```diff
@@ -42,24 +40,16 @@ export const createAlertAttachmentUserActionBuilder = ({
   onShowAlertDetails,
 }: BuilderArgs): ReturnType<UserActionBuilder> => ({
   build: () => {
-    const alertId = getFirstItem(comment.alertId);
-    const alertIndex = getFirstItem(comment.index);
+    const alertId = getNonEmptyField(comment.alertId);
+    const alertIndex = getNonEmptyField(comment.index);
 
-    if (isEmpty(alertId)) {
+    if (!alertId || !alertIndex) {
       return [];
     }
 
-    const ruleId =
-      comment?.rule?.id ??
-      alertData[alertId]?.signal?.rule?.id?.[0] ??
-      get(alertData[alertId], ALERT_RULE_UUID)[0] ??
```

Comment:
I found another issue here, in my testing this would result in a regular string not a string array. Which was either causing a crash in the UI or it would display the first letter of the rule id.

Reply:
@peteharverson managed to crush it 🙂

Reply:
Confirmed this fixes the crash I was previously seeing with the test case described in this PR.